packages/application-shell/src/components/application-shell/application-shell.tsx

```diff
@@ -149,6 +155,17 @@ export const RestrictedApplication = <
   // However, every route change will trigger a re-render. This is probably
   // ok-ish but we might want to look into a more performant solution.
   const location = useLocation();
+
+  const notificationsGlobalRef = useRef<HTMLDivElement>(null);
+  const notificationsPageRef = useRef<HTMLDivElement>(null);
+  const layoutRefs = useRef<{
```

Comment:
This is auxiliary and ends up in the portals container so that there we can pass it on to the mutation observer.